src/main/java/com/amazonaws/encryptionsdk/keyrings/MultiKeyring.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except
+ * in compliance with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.amazonaws.encryptionsdk.keyrings;
+
+import com.amazonaws.encryptionsdk.EncryptedDataKey;
+import com.amazonaws.encryptionsdk.exception.AwsCryptoException;
+import com.amazonaws.encryptionsdk.exception.CannotUnwrapDataKeyException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.lang3.Validate.isTrue;
+
+/**
+ * A keyring which combines other keyrings, allowing one OnEncrypt or OnDecrypt call to
+ * modify the encryption or decryption materials using more than one keyring.
+ */
+class MultiKeyring implements Keyring {
+
+    final Keyring generatorKeyring;
+    final List<Keyring> childrenKeyrings;
+
+    MultiKeyring(Keyring generatorKeyring, List<Keyring> childrenKeyrings) {
+        this.generatorKeyring = generatorKeyring;
+        this.childrenKeyrings = childrenKeyrings == null ? emptyList() : unmodifiableList(childrenKeyrings);
```

Comment:
Might be worth making a defensive copy of `childrenKeyrings`. `unmodifiableList` keeps you from modifying it, not anyone else.

Reply:
Done. I'll make this change in the KmsKeyring too